smithy-aws-protocol-tests/model/query/xml-maps.smithy

```diff
@@ -0,0 +1,211 @@
+}
+
+map XmlMapsOutputMap {
+    key: String,
+    value: GreetingStruct
+}
+
+// This example tests maps with @xmlName on members.
+operation XmlMapsXmlName() -> XmlMapsXmlNameOutput
+
+apply XmlMapsXmlName @httpResponseTests([
+    {
+        id: "QueryQueryXmlMapsXmlName",
+        description: "Serializes XML lists",
+        protocol: "aws.query",
+        code: 200,
+        body: """
+              <XmlMapsXmlNameResponse xmlns="https://example.com/">
+                  <XmlMapsXmlNameResult>
+                      <myMap>
+                          <entry>
+                              <Name>foo</Name>
+                              <Setting>
+                                  <hi>there</hi>
+                              </Setting>
+                          </entry>
+                          <entry>
+                              <Name>baz</Name>
+                              <Setting>
+                                  <hi>bye</hi>
+                              </Setting>
+                          </entry>
+                      </myMap>
+                  </XmlMapsXmlNameResult>
+              </XmlMapsXmlNameResponse>
+              """,
+        bodyMediaType: "application/xml",
+        headers: {
+            "Content-Type": "text/xml"
+        },
+        params: {
+            myMap: {
+                foo: {
+                    hi: "there"
+                },
+                baz: {
+                    hi: "bye"
+                }
+            }
+        }
+    }
+])
+
+structure XmlMapsXmlNameOutput {
+    myMap: XmlMapsXmlNameOutputMap,
+}
+
+map XmlMapsXmlNameOutputMap {
+    @xmlName("Attribute")
+    key: String,
+
+    @xmlName("Setting")
+    value: GreetingStruct
+}
+
+/// Flattened maps
+operation FlattenedXmlMap() -> FlattenedXmlMapOutput
+
+apply FlattenedXmlMap @httpResponseTests([
+    {
+        id: "QueryQueryFlattenedXmlMap",
+        description: "Serializes flattened XML maps in responses",
+        protocol: "aws.query",
+        code: 200,
+        body: """
+              <FlattenedXmlMapResponse xmlns="https://example.com/">
+                  <FlattenedXmlMapResult>
+                      <myMap>
+                          <key>foo</key>
+                          <value>Foo</value>
+                      </myMap>
+                      <myMap>
+                          <key>baz</key>
+                          <value>Baz</value>
+                      </myMap>
+                  </FlattenedXmlMapResult>
+              </FlattenedXmlMapResponse>""",
+        bodyMediaType: "application/xml",
+        headers: {
+            "Content-Type": "text/xml"
+        },
+        params: {
+            myMap: {
+                foo: "Foo",
+                baz: "Baz"
+            }
+        }
+    }
+])
+
+structure FlattenedXmlMapOutput {
+    @xmlFlattened
+    myMap: FooEnumMap,
+}
+
+/// Flattened maps with @xmlName
+operation FlattenedXmlMapWithXmlName() -> FlattenedXmlMapWithXmlNameOutput
```

Comment:
Closing tags for `</myMap>` should be `</KVP>` for both traits (below) applied to this operation.